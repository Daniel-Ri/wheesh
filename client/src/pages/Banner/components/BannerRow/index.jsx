import { Button, Dialog } from '@mui/material';
import PropTypes from 'prop-types';
import { FormattedMessage, injectIntl } from 'react-intl';

import { useNavigate } from 'react-router-dom';
import ModeEditOutlinedIcon from '@mui/icons-material/ModeEditOutlined';
import DeleteOutlineOutlinedIcon from '@mui/icons-material/DeleteOutlineOutlined';
import config from '@config/index';
import { useMediaQuery } from 'react-responsive';

import { useState } from 'react';
import { useDispatch } from 'react-redux';
import toast from 'react-hot-toast';
import { deleteBanner } from '@pages/Banner/actions';
import classes from './style.module.scss';

const BannerRow = ({ banner, intl: { formatMessage } }) => {
  const [open, setOpen] = useState(false);
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const isMobile = useMediaQuery({ query: '(max-width: 768px)' });

  const handleClose = () => {
    setOpen(false);
  };

  const handleSuccessDelete = () => {
    toast.success(formatMessage({ id: 'app_success_delete_banner' }));
  };

  const handleErrorDelete = (errorMsg) => {
    toast.error(errorMsg);
  };

  const handleDeleteBanner = () => {
    dispatch(deleteBanner(banner.id, handleSuccessDelete, handleErrorDelete));
  };

  return (
    <>
      {!isMobile ? (
        <tr data-testid="BannerRowDesktop">
          <td>
            <div className={classes.image}>
              <img src={`${config.api.host}${banner.imageDesktop}`} alt="" />
            </div>
          </td>
          <td>
            <div className={classes.image}>
              <img src={`${config.api.host}${banner.imageMobile}`} alt="" />
            </div>
          </td>
          <td className={classes.buttons}>
            <Button
              data-testid="ChangeBannerButton"
              variant="contained"
              className={classes.btn}
              onClick={() => navigate(`/changeBanner/${banner.id}`)}
            >
              <ModeEditOutlinedIcon />
            </Button>
            <Button
              data-testid="DeleteBannerButton"
              variant="contained"
              className={`${classes.btn} ${classes.delete}`}
              onClick={() => setOpen(true)}
            >
              <DeleteOutlineOutlinedIcon />
            </Button>
          </td>
        </tr>
      ) : (
        <>
          <tr data-testid="BannerRowMobile">
            <th>Desktop</th>
            <td>
              <div className={classes.image}>
                <img src={`${config.api.host}${banner.imageDesktop}`} alt="" />
              </div>
            </td>
            <td className={classes.editCell}>
              <Button
                variant="contained"
                className={classes.btn}
                onClick={() => navigate(`/changeBanner/${banner.id}`)}
              >
                <ModeEditOutlinedIcon />
              </Button>
            </td>
          </tr>
          <tr>
            <th>Mobile</th>
            <td>
              <div className={classes.imageMobile}>
                <img src={`${config.api.host}${banner.imageMobile}`} alt="" />
              </div>
            </td>
            <td className={classes.deleteCell}>
              <Button variant="contained" className={`${classes.btn} ${classes.delete}`} onClick={() => setOpen(true)}>
                <DeleteOutlineOutlinedIcon />
              </Button>
            </td>
          </tr>
        </>
      )}
      <Dialog data-testid="Dialog" open={open} onClose={handleClose}>
        <div className={classes.dialog}>
          <div className={classes.content}>
            <h2>
              <FormattedMessage id="app_verification" />
            </h2>
            <div className={classes.message}>
              <FormattedMessage id="app_sure_deleting_the_banner" />
            </div>
          </div>
          <div className={classes.buttons}>
            <Button data-testid="Cancel" variant="outlined" className={classes.cancel} onClick={handleClose}>
              <FormattedMessage id="app_cancel" />
            </Button>
            <Button data-testid="Confirm" variant="contained" className={classes.confirm} onClick={handleDeleteBanner}>
              <FormattedMessage id="app_confirm" />
            </Button>
          </div>
        </div>
      </Dialog>
    </>
  );
};

BannerRow.propTypes = {
  banner: PropTypes.object.isRequired,
  intl: PropTypes.object,
};

export default injectIntl(BannerRow);
