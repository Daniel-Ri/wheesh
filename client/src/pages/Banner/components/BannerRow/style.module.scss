@import '@styles/base';

.image {
  width: 100%;
  height: 7rem;
}

.imageMobile {
  width: 50%;
  height: 7rem;
  margin: auto;
}

img {
  width: 100%;
  height: 100%;
  object-fit: fill;
}

.buttons {
  .btn {
    margin: 0.5rem auto;
    width: 2rem;
    height: 2rem;
    padding: 0;
    @include directionX;
    align-items: center;
    justify-content: center;
  }

  .btn.delete {
    background-color: $color-strawberry-red;
  }
}

.btn {
  margin: 0.5rem auto;
  width: 2rem;
  height: 2rem;
  padding: 0;
  @include directionX;
  align-items: center;
  justify-content: center;
}

.btn.delete {
  background-color: $color-strawberry-red;
}

.btn.delete:hover {
  background-color: $color-strawberry-red;
}

td.editCell {
  border-bottom-width: 0 !important;
}

td.deleteCell {
  border-top-width: 0 !important;
}

.dialog {
  width: 16rem;
  padding: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  .content {
    display: flex;
    flex-direction: column;
    gap: 1rem;

    h2 {
      text-align: center;
      color: $color-marine-blue;
      font-size: 120%;
    }
  }

  .buttons {
    margin-top: 1rem;
    margin-bottom: 0.25rem;
    display: flex;
    justify-content: flex-end;
    gap: 1rem;

    .cancel {
      text-transform: capitalize;
    }

    .confirm {
      text-transform: capitalize;
      background-color: $color-strawberry-red;
    }
  }
}