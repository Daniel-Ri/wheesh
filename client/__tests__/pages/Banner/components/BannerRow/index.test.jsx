import { render } from '@utils/testHelper';
import BannerRow from '@pages/Banner/components/BannerRow';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}));

let wrapper;

const mockProps = {
  banner: {
    id: 1,
    imageDesktop: 'public/Frame 1 - Desktop.png',
    imageMobile: 'public/Frame 1 - Mobile.png',
  },
};

beforeEach(() => {
  wrapper = render(<BannerRow banner={mockProps.banner} />);
});

describe('Component BannerRow of Banner component', () => {
  test('Renders BannerRow component', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('BannerRowDesktop')).toBeInTheDocument();
  });
});
