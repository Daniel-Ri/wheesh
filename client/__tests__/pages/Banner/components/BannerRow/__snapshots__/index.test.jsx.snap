// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component BannerRow of Banner page Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style=""
  >
    <div>
      <tr
        data-testid="BannerRowDesktop"
      >
        <td>
          <div
            class="image"
          >
            <img
              alt=""
              src="http://localhost:3300/api/public/Frame 1 - Desktop.png"
            />
          </div>
        </td>
        <td>
          <div
            class="image"
          >
            <img
              alt=""
              src="http://localhost:3300/api/public/Frame 1 - Mobile.png"
            />
          </div>
        </td>
        <td
          class="buttons"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium btn css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="ChangeBannerButton"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ModeEditOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 21h3.75L17.81 9.94l-3.75-3.75L3 17.25V21zm2-2.92 9.06-9.06.92.92L5.92 19H5v-.92zM18.37 3.29a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41l-2.34-2.34z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium btn delete css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="DeleteBannerButton"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteOutlineOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5-1-1h-5l-1 1H5v2h14V4h-3.5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr
      data-testid="BannerRowDesktop"
    >
      <td>
        <div
          class="image"
        >
          <img
            alt=""
            src="http://localhost:3300/api/public/Frame 1 - Desktop.png"
          />
        </div>
      </td>
      <td>
        <div
          class="image"
        >
          <img
            alt=""
            src="http://localhost:3300/api/public/Frame 1 - Mobile.png"
          />
        </div>
      </td>
      <td
        class="buttons"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium btn css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="ChangeBannerButton"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ModeEditOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 21h3.75L17.81 9.94l-3.75-3.75L3 17.25V21zm2-2.92 9.06-9.06.92.92L5.92 19H5v-.92zM18.37 3.29a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41l-2.34-2.34z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium btn delete css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="DeleteBannerButton"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DeleteOutlineOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5-1-1h-5l-1 1H5v2h14V4h-3.5z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
